presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.h

```diff
@@ -26,19 +26,17 @@ class PartitionAndSerializeNode : public velox::core::PlanNode {
       const velox::core::PlanNodeId& id,
       std::vector<velox::core::TypedExprPtr> keys,
       int numPartitions,
-      velox::RowTypePtr outputType,
-      velox::core::PlanNodePtr source)
+      velox::core::PlanNodePtr source,
+      velox::core::PartitionFunctionFactory partitionFunctionFactory)
       : velox::core::PlanNode(id),
-        keys_{std::move(keys)},
-        numPartitions_{numPartitions},
-        outputType_{std::move(outputType)},
-        sources_{std::move(source)} {
-    VELOX_USER_CHECK(
-        velox::ROW(
-            {"partition", "data"}, {velox::INTEGER(), velox::VARBINARY()})
-            ->equivalent(*outputType_));
+        keys_(std::move(keys)),
+        numPartitions_(numPartitions),
+        outputType_(velox::ROW(
```

Comment:
I suggest to keep the outputType parameter to allow the caller to specify custom column names. All we need is to enforce that outputType has 2 columns of type INTEGER and VARBINARY.

Reply:
In what scenarios will we need the column name to be customly specified?